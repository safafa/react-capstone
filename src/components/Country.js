/* eslint-disable react/prop-types */
import React from 'react';
import { Card } from 'react-bootstrap';
import { BsArrowRightCircle } from 'react-icons/bs';

const Country = ({ country }) => {
  const page = `/details/${country.id}`;
  return (
    <a className="w-50 text-decoration-none link-dark text-center" href={page}>
      <Card>
        <div className="d-flex justify-content-between align-items-center px-3 pt-2">
          <Card.Title>{country.name}</Card.Title>
          <BsArrowRightCircle />
        </div>
        <Card.Body>
          <Card.Text>
            {country.today_confirmed}
          </Card.Text>
        </Card.Body>
      </Card>
    </a>
  );
};

export default Country;
