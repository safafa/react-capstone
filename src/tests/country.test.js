import renderer from 'react-test-renderer';
import Country from '../components/Country';

describe('Country component', () => {
  test('Snapshot test', () => {
    const tree = renderer.create(
      <Country country={{ id: 'test', name: 'test', today_confirmed: '12314' }} />,
    )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
