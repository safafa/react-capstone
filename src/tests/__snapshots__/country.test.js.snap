// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country component Snapshot test 1`] = `
<a
  className="w-50 text-decoration-none link-dark text-center text-light"
  href="/details/test"
>
  <div
    className="bg-pink border-0 m-1 cardo card"
  >
    <div
      className="d-flex justify-content-between align-items-center px-3 pt-2"
    >
      <div
        className="title-f card-title h5"
      >
        test
      </div>
      <svg
        className="arrow"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <div
      className="card-body"
    >
      <p
        className="card-text"
      >
        12314
      </p>
    </div>
  </div>
</a>
`;
